qcore/nhm.py

```diff
@@ -29,53 +58,157 @@ class NHMFault:
+        slip_rate = sample_trunc_norm_dist(self.slip_rate, self.slip_rate_sigma)
+        coupling_coeff = sample_trunc_norm_dist(
+            self.coupling_coeff, self.coupling_coeff_sigma
+        )
+        width = (dbot - dtop) / np.sin(np.radians(dip))
+
+        if mw_area_scaling:
+            if self.tectonic_type == "VOLCANIC" or (
+                self.tectonic_type == "ACTIVE_SHALLOW"
+                and self.fault_type == "NORMAL_FAULTING"
+            ):
+                mw_scaling_rel = mag_scaling.MagnitudeScalingRelations.VILLAMORETAL2007
+            elif self.tectonic_type == "SUBDUCTION_INTERFACE":
+                ### SUBDUCTION INTERFACE RELATION IS TO BE CONFIRMED BY BB leaving equations here for reference
+
+                mw_old = 4.441 + 0.846 * np.log10(length * width)
+                mom = (
+                    self.recur_int_median
+                    * MU
+                    * length
+                    * width
+                    * self.slip_rate
+                    * 10e3
+                    * self.coupling_coeff
+                )
+                mw = (np.log10(mom) - 9.05) / 1.5
+
+                mw_scaling_rel = mag_scaling.MagnitudeScalingRelations.SKARLATOUDIS2016
+            elif self.fault_type == "PLATE_BOUNDARY":
+                mw_scaling_rel = mag_scaling.MagnitudeScalingRelations.HANKSBAKUN2002
+            elif self.fault_type == "OTHER_CRUSTAL_FAULTING":
+                mw_scaling_rel = mag_scaling.MagnitudeScalingRelations.STIRLING2008
+            else:
+                raise (
+                    ValueError,
+                    f"Invalid combination of tectonic type: {self.tectonic_type} and fault type: {self.fault_type}",
+                )
 
-        self.name = rows[0]
-        self.tectonic_type, self.fault_type = rows[1].split()
-        self.length, self.length_sigma = str2floats(rows[2])
-        self.dip, self.dip_sigma = str2floats(rows[3])
-        self.dip_dir = float(rows[4])
-        self.rake = float(rows[5])
-        self.dbottom, self.dbottom_sigma = str2floats(rows[6])
-        self.dtop, self.dtop_min, self.dtop_max = str2floats(rows[7])
-        self.slip_rate, self.slip_rate_sigma = str2floats(rows[8])
-        self.coupling_coeff, self.coupling_coeff_sigma = str2floats(rows[9])
-        self.mw, self.recur_int_median = str2floats(rows[10])
-        self.trace = np.array(list(map(float, " ".join(rows[12:]).split()))).reshape((-1, 2))
-        # TODO: add x y z fault plane data as in SRF info
-        # TODO: add leonard mw function
-
-
-def load_nhm(nhm_path: str, skiprows: int=15):
+            mw_median, mw_sigma = mag_scaling.lw_2_mw_sigma_scaling_relation(
+                length, width, mw_scaling_rel, self.rake
+            )
+
+            if mw_perturbation:
+                mw = sample_trunc_norm_dist(mw_median, mw_sigma)
+            else:
+                mw = mw_median
+
+        moment = 10 ** (9.05 + 1.5 * mw)
```

Comment:
Where did this magnitude-moment relationship come from? It seems to differ from the mag2mom function in mag_scaling.py 10^(9.05+1.5*Mw) vs 10^(16.05+1.5*Mw) (I think)

Reply:
Correct, that one is incorrect (dyne-cms) while this one is moment metres. As part of the scaling relation PR, I've added comments / these functions.
![image](https://user-images.githubusercontent.com/25143301/118886152-af9eb900-b94c-11eb-99ef-cf857ebc7c33.png)